### PR TITLE
List test datasets and update PR template

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
Adds a `list_datasets` function to list datasets in the xclim-testdata repo using github's REST API. 
Update the PR template because I want to list things in the first section but it's already a list.

* **Does this PR introduce a breaking change?** 
Nope

* **Other information**:
The API calls are unauthenticated and thus rate-limited to 60 per hour (per id). The function makes 1 main + 1 per subdirectory, so 3 calls and you're blocked. This is not really a problem, this is just a convenience function, but I marked the test as "slow" so it's only executed in one build.

### CODE REVIEW

File: xclim/testing/tests/test_utils.py [hunk 1]
```diff
@@ -107,10 +107,10 @@ def test_open_testdata():
...
+    assert list(out.columns) == ["size", "url"]
     np.testing.assert_allclose(
-        out.loc[
-            "cmip6/o3_Amon_GFDL-ESM4_historical_r1i1p1f1_gr1_185001-194912.nc"
-        ].size,
-        0.825216,
+        out.loc["cmip6/o3_Amon_GFDL-ESM4_historical_r1i1p1f1_gr1_185001-194912.nc"][
+            "size"
+        ],
+        845.021484375,
```
Comment (Zeitsperre) [inline xclim/testing/tests/test_utils.py]: What changed here? Those figures are quite the departure.

Comment (aulemahal) [reply xclim/testing/tests/test_utils.py]: haha, I wrote the test in MB, but then I changed to code to return KB but forgot to change the value here!

**Comment by aulemahal** [comment]
The 2 CodeFactor issues are non-issues (because we already know all URLs are https, there are no user-provided urls), thus this is ready!

**Review by Zeitsperre — APPROVED** [review]
Looks great! Good idea!

**Pull request merged**